_posts/2023-09-07-develop-functions-locally.md

```diff
@@ -0,0 +1,278 @@
+---
+title: "The faster way to iterate on your OpenFaaS functions"
+description: "Learn how the local-run and watch features of faas-cli can help you iterate on functions faster."
+date: 2023-09-07
+categories:
+- faas-cli
+- openfaas
+- functions
+- devex
+dark_background: true
+author_staff_member: han
+image: "/images/2023-09-develop-functions-locally/background.png"
+hide_header_image: true
+---
+
+Learn how the local-run and watch features of faas-cli can help you iterate on functions faster.
+
+This is what the typical development lifecycle of an OpenFaaS function looks like:
+
+- Make code changes
+- Build the function image
+- Push the image to a container registry
+- Deploy the function
+
+The faas-cli tries to minimize the amount of manual steps by bundling some of these actions into a single command. For example, running `faas-cli up` will build, push and deploy functions.
+
+The disadvantage of this workflow is that it can introduce some delay as you will have to wait for the function image to be pushed to a registry, pulled into a node and started up each time you make changes to your code. 
+
+Thanks to some recent work from the community a couple of new features were added to the faas-cli to further improve the development experience.
+
+A new command, `faas-cli local-run` was added. The command runs a function as a docker container directly on your machine. Since the function spins up pretty much instantly this allows you to test your function locally while quickly iterating over your code.
```

Comment:
`docker container` - just container, and Docker is a proper noun so should be capitalised when it's not the command